x-pack/legacy/plugins/oss_telemetry/server/lib/collectors/csp/csp_collector.test.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import sinon from 'sinon';
+import {
+  DEFAULT_CSP_RULES,
+  DEFAULT_CSP_STRICT,
+  DEFAULT_CSP_WARN_LEGACY_BROWSERS,
+} from '../../../../../../../../src/legacy/server/csp';
+import {
+  getMockCallWithInternal,
+  getMockKbnServer,
+  getMockTaskFetch,
+} from '../../../../test_utils';
+import { createCspCollector } from './csp_collector';
+
+test('fetches whether strict mode is enabled', async () => {
+  const { collector, mockConfig } = setupCollector();
+
+  expect((await collector.fetch()).strict).toEqual(true);
+
+  mockConfig.get.withArgs('csp.strict').returns(false);
+  expect((await collector.fetch()).strict).toEqual(false);
+});
+
+test('fetches whether the legacy browser warning is enabled', async () => {
+  const { collector, mockConfig } = setupCollector();
+
+  expect((await collector.fetch()).warnLegacyBrowsers).toEqual(true);
+
+  mockConfig.get.withArgs('csp.warnLegacyBrowsers').returns(false);
+  expect((await collector.fetch()).warnLegacyBrowsers).toEqual(false);
+});
+
+test('fetches whether the csp rules have been changed or not', async () => {
+  const { collector, mockConfig } = setupCollector();
+
+  expect((await collector.fetch()).rulesChangedFromDefault).toEqual(false);
+
+  mockConfig.get.withArgs('csp.rules').returns(['not', 'default']);
+  expect((await collector.fetch()).rulesChangedFromDefault).toEqual(true);
+});
+
+test('does not include raw csp.rules', async () => {
```

Comment:
question: should we consider using jest's snapshots to ensure that only the properties we expect are returned? if we were to expose the actual rules under a property named differently than "rules", like "cspRules" this test wouldn't catch it.

Reply:
That's a really good point. My goal here was to make it so we didn't break tests just by collecting a new data point, which shouldn't be considered a breaking change and thus shouldn't break the test, but for sensitive data like this perhaps it's best to err on the side of caution.